### PR TITLE
26527 add core policy wsdl trace

### LINKED ISSUE #26527: Enhance trace in CXF core, policy and WSDL modules
The existing CXF trace in the core (com.ibm.ws.org.apache.cxf.cxf.core.3.2),  policy (com.ibm.ws.org.apache.cxf.cxf.rt.ws.policy.3.2) and WSDL (com.ibm.ws.org.apache.cxf.cxf.rt.wsdl.3.2) modules is not sufficient to debug customer problems. The trace in these modules need to be increased to make it easier to debug problems.

### PR DESCRIPTION
**Pull request by sawadood**
fixes #26527

### CODE REVIEW

**Comment by sawadood** [comment]
#build

**Comment by sawadood** [comment]
#build with new commits

**Comment by sawadood** [comment]
#build - start a new PB

**Comment by sawadood** [comment]
#build again

**Comment by sawadood** [comment]
#build

**Comment by sawadood** [comment]
#build

**Comment by sawadood** [comment]
#build

**Comment by sawadood** [comment]
#build

**Comment by sawadood** [comment]
Current PB running at

**Comment by sawadood** [comment]
Current PB running at:   https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=_5RzxcNWFEe6vioT4Ep3iAw

**Comment by sawadood** [comment]
There is one failure in my last PB in com.ibm.ws.springboot.fat20_fat FAT, which is unrelated to my changes.

**Review by jim-krueger — CHANGES_REQUESTED** [review]
Still reviewing, but you need to change all of the parts you've changed to have 2024 in the copyright.

File: dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/interceptor/StaxInInterceptor.java [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+                        }
+                    }
+                }
+            }
+	    // Liberty Change start
+	    LOG.finest("XMLStreamReader before configureReader: " + xreader.getClass().getCanonicalName());
+            xreader = StaxUtils.configureReader(xreader, message);
+	    LOG.finest("XMLStreamReader after configureReader: " + xreader.getClass().getCanonicalName());
+	    // Liberty Change end
+	    LOG.finest("XMLStreamReader before configureReader: " + xreader.getClass().getCanonicalName());
```
Comment (neuwerk) [inline dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/interceptor/StaxInInterceptor.java]: Duplicated trace lines
```suggestion
```

File: dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/service/model/MessagePartInfo.java [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+
+import javax.xml.namespace.QName;
+
+import org.apache.ws.commons.schema.XmlSchemaAnnotated;
+import org.apache.ws.commons.schema.XmlSchemaElement;
+import java.util.logging.Logger;
+import org.apache.cxf.common.logging.LogUtils;
+import java.util.logging.Level;
+
+public final class MessagePartInfo extends AbstractPropertiesHolder implements NamedItem {
```
Comment (neuwerk) [inline dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/service/model/MessagePartInfo.java:30]: I think we shouldn't let trace injection be added to this class. It's one of the core pieces of CXF, and so trace injection will be logging this class like crazy. Can you add this class to the list of excludes in the `bnd.bnd`

File: dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/staxutils/StaxUtils.java [hunk 3]
```diff
@@ -243,6 +243,7 @@ private static int getInteger(String prop, int def) {
             }
             return i;
         } catch (Throwable t) {
+            LOG.finest("getInteger: Ignoring exception: " + t);  // Liberty Change
```
Comment (neuwerk) [inline dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/staxutils/StaxUtils.java:246]: Same for this file. I think its fine to have the added trace logging, but make sure the trace injection is turned off for this class.

File: dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/common/logging/LogUtils.java [hunk 4]
```diff
@@ -165,9 +166,8 @@ public static void setLoggerClass(Class<? extends AbstractDelegatingLogger> cls)
      * @return an appropriate Logger
      */
     public static Logger getLogger(Class<?> cls) {
-        //Liberty Change for CXF Begain
-        return createLogger(cls, null, cls.getName() + getClassLoader(cls));
-        //Liberty Change for CXF End
+	Logger logger = createLogger(cls, null, cls.getName() + getClassLoader(cls));
```
Comment (neuwerk) [inline dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/common/logging/LogUtils.java:169]: I'm fine with the added trace for this file, but will you make sure that the trace injection is turned off for this class in the `bnd.bnd`?

**Review by neuwerk — CHANGES_REQUESTED** [review]
See comments

File: dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/common/logging/LogUtils.java (hunk 4 above)
Comment (sawadood) [reply dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/common/logging/LogUtils.java:169]: I removed all trace from LogUtils.java with the commit  "Remove trace from LogUtils.java ".

File: dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/interceptor/StaxInInterceptor.java (hunk 1 above)
Comment (sawadood) [reply dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/interceptor/StaxInInterceptor.java]: Removed duplicate trace statement.

File: dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/service/model/MessagePartInfo.java (hunk 2 above)
Comment (sawadood) [reply dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/service/model/MessagePartInfo.java:30]: I agree, disabled trace injection for this class.

File: dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/common/logging/LogUtils.java (hunk 4 above)
Comment (sawadood) [reply dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/common/logging/LogUtils.java:169]: I already removed all trace from this class.

File: dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/staxutils/StaxUtils.java (hunk 3 above)
Comment (sawadood) [reply dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/staxutils/StaxUtils.java:246]: Disabled trace injection for StaxUtils.class.

**Comment by sawadood** [comment]
#build Another PB after resolving review comments.

File: dev/com.ibm.ws.jaxws.ejb_fat/publish/servers/com.ibm.ws.jaxws.ejb.fat.ejbwsprovider/bootstrap.properties [hunk 5]
```diff
@@ -1 +1,2 @@
-bootstrap.include=../testports.properties
\ No newline at end of file
+bootstrap.include=../testports.properties
+com.ibm.ws.logging.trace.specification=org.apache.cxf.*=all=enabled:com.ibm.ws.jaxws.*=all=enabled
```
Comment (jim-krueger) [inline dev/com.ibm.ws.jaxws.ejb_fat/publish/servers/com.ibm.ws.jaxws.ejb.fat.ejbwsprovider/bootstrap.properties]: Is it necessary to have this trace enabled all of the time and increase the output?  Could it maybe be enabled only when debugging?

File: dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/attachment/AttachmentUtil.java [hunk 6]
```diff
@@ -96,13 +103,17 @@ public DataContentHandler createDataContentHandler(String mimeType,
                 DataSource ds) {
             DataContentHandler dch = super.createDataContentHandler(mimeType);
             if (dch == null) {
+	        if (LOG.isLoggable(Level.FINE)) { //Liberty Change Start
+	           LOG.fine("createDataContentHandler using DEFAULT_COMMAND_MAP for DataSource: " + ds.getName());
```
Comment (jim-krueger) [inline dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/attachment/AttachmentUtil.java]: Should ds be null checked to avoid an NPE here?

File: dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/interceptor/MessageSenderInterceptor.java [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+import java.util.logging.Logger;
+
+/**
+ * Takes the Conduit from the exchange and sends the message through it.
+ */
+public class MessageSenderInterceptor extends AbstractPhaseInterceptor<Message> {
+    private static final ResourceBundle BUNDLE = BundleUtils.getBundle(MessageSenderInterceptor.class);
+    private MessageSenderEndingInterceptor ending = new MessageSenderEndingInterceptor();
+
+    private static final Logger LOG = LogUtils.getL7dLogger(MessageSenderInterceptor.class);  // Liberty Change
```
Comment (jim-krueger) [inline dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/interceptor/MessageSenderInterceptor.java]: Why is this added if there is no usage of LOG?   Are there other Liberty changes in this class?

File: dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/interceptor/OneWayProcessorInterceptor.java [hunk 8]
```diff
@@ -0,0 +1,181 @@
...
+                            });
+                        //wait a few milliseconds for the background thread to start processing
+                        //Mostly just to make an attempt at keeping the ordering of the
+                        //messages coming in from a client.  Not guaranteed though.
+                        lock.wait(20L);
+                    }
+                } catch (RejectedExecutionException e) {
+                    LOG.warning(
+                        "Executor queue is full, run the oneway invocation task in caller thread."
+                        + "  Users can specify a larger executor queue to avoid this.");
```
Comment (jim-krueger) [inline dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/interceptor/OneWayProcessorInterceptor.java:153]: Should this have a Liberty change flag?

File: dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/interceptor/StaxOutEndingInterceptor.java [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+        super(Phase.PRE_STREAM_ENDING);
+        getAfter().add(AttachmentOutInterceptor.AttachmentOutEndingInterceptor.class.getName());
+        this.outStreamHolder = outStreamHolder;
+        this.writerHolder = writerHolder;
+    }
+
+    public void handleMessage(Message message) {
+        try {
+            XMLStreamWriter xtw = message.getContent(XMLStreamWriter.class);
+	    LOG.finest("XMLStreamWriter class: " + xtw.getClass().getCanonicalName()); // Liberty Change
```
Comment (jim-krueger) [inline dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/interceptor/StaxOutEndingInterceptor.java]: xtw should be  null checked to avoid an NPE prior to this LOG?

File: dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/interceptor/StaxOutInterceptor.java [hunk 10]
```diff
@@ -0,0 +1,207 @@
...
+                            os = setupOutputStream(os);
+                            xwriter = factory.createXMLStreamWriter(os, encoding);
+                        } else {
+                            xwriter = factory.createXMLStreamWriter(writer);
+                        }
+                    }
+                }
+            }
+
+	    LOG.finest("XMLStreamWriter class: " + xwriter.getClass().getCanonicalName()); // Liberty Change
```
Comment (jim-krueger) [inline dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/interceptor/StaxOutInterceptor.java]: Does xwriter need to be null checked?

File: dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/staxutils/StaxUtils.java [hunk 11]
```diff
@@ -390,6 +392,7 @@ private static boolean setProperty(XMLInputFactory f, String p, Object o) {
             f.setProperty(p,  o);
             return true;
         } catch (Throwable t) {
+	    LOG.finest("setProperty: Ignoring exception: " + t);
```
Comment (jim-krueger) [inline dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/staxutils/StaxUtils.java]: Liberty change flag?

File: dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/staxutils/StaxUtils.java [hunk 12]
```diff
@@ -594,6 +597,7 @@ public static void copy(Source source, OutputStream os) throws XMLStreamExceptio
             try {
                 writer.flush();
             } catch (XMLStreamException ex) {
+	        LOG.finest("copy: Ignoring XMLStreamException: " + ex);
```
Comment (jim-krueger) [inline dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/staxutils/StaxUtils.java]: Liberty change flag?

File: dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/staxutils/StaxUtils.java [hunk 13]
```diff
@@ -624,10 +628,12 @@ public static void copy(Source source, XMLStreamWriter writer) throws XMLStreamE
                             reader.setFeature("http://xml.org/sax/features/namespaces", true);
                         } catch (Throwable t) {
                             //ignore
+	        	    LOG.finest("copy: Ignoring exception 1: " + t);
```
Comment (jim-krueger) [inline dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/staxutils/StaxUtils.java]: Liberty change flag?

File: dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/staxutils/StaxUtils.java [hunk 14]
```diff
@@ -624,10 +628,12 @@ public static void copy(Source source, XMLStreamWriter writer) throws XMLStreamE
...
                         }
                         try {
                             reader.setProperty("http://xml.org/sax/properties/lexical-handler", ch);
                         } catch (Throwable t) {
+	        	    LOG.finest("copy: Ignoring exception 2: " + t);
```
Comment (jim-krueger) [inline dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/staxutils/StaxUtils.java]: Liberty change flag?

File: dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/transport/ChainInitiationObserver.java [hunk 15]
```diff
@@ -187,26 +179,31 @@ protected void setExchangeProperties(Exchange exchange, Message m) {
...
                     } catch (URISyntaxException e) {
+	                 LOG.fine("setExchangeProperties: Ignoring URISyntaxException: " + e);  // Liberty Change
                         // do nothing
                     }
                     endpointInfo.setProperty("URI", wsdlDescription);
                 }
                 exchange.put(Message.WSDL_DESCRIPTION, wsdlDescription);
             }
         } else {
+	    LOG.fine("setExchangeProperties: Setting Service.class to null");
```
Comment (jim-krueger) [inline dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/transport/ChainInitiationObserver.java]: Liberty change flag?

File: dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/workqueue/SynchronousExecutor.java [hunk 16]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+    public static SynchronousExecutor getInstance() {
+        return INSTANCE;
+    }
+
+    public static boolean isA(Executor executor) {
+        return executor == INSTANCE;
+    }
+}
```
Comment (jim-krueger) [inline dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/workqueue/SynchronousExecutor.java:47]: What is the Liberty change in this class?

File: dev/com.ibm.ws.org.apache.cxf.cxf.rt.ws.policy.3.2/src/org/apache/cxf/ws/policy/PolicyEngineImpl.java [hunk 17]
```diff
@@ -677,6 +686,9 @@ public boolean supportsAlternative(Collection<? extends PolicyComponent> alterna
                             }
                             return true;
                         } else {
+                            if (doLog) {
+                               LOG.fine("ignoreUnsupportedPolicy is not set, returning false 1");
+			    }
```
Comment (jim-krueger) [inline dev/com.ibm.ws.org.apache.cxf.cxf.rt.ws.policy.3.2/src/org/apache/cxf/ws/policy/PolicyEngineImpl.java:691]: Liberty change flag?

File: dev/com.ibm.ws.org.apache.cxf.cxf.rt.ws.policy.3.2/src/org/apache/cxf/ws/policy/PolicyEngineImpl.java [hunk 18]
```diff
@@ -694,12 +706,18 @@ public boolean supportsAlternative(Collection<? extends PolicyComponent> alterna
 			       }
                                return true;
                             } else {
+                               if (doLog) {
+                                  LOG.fine("ignoreUnsupportedPolicy is not set, returning false 2");
```
Comment (jim-krueger) [inline dev/com.ibm.ws.org.apache.cxf.cxf.rt.ws.policy.3.2/src/org/apache/cxf/ws/policy/PolicyEngineImpl.java:710]: Liberty change flag?

File: dev/com.ibm.ws.org.apache.cxf.cxf.rt.ws.policy.3.2/src/org/apache/cxf/ws/policy/PolicyEngineImpl.java [hunk 19]
```diff
@@ -694,12 +706,18 @@ public boolean supportsAlternative(Collection<? extends PolicyComponent> alterna
...
+			       }
                                return false;
                             } // Liberty Change End
                         }
                     }
                 }
             } else {
+                if (doLog) {
```
Comment (jim-krueger) [inline dev/com.ibm.ws.org.apache.cxf.cxf.rt.ws.policy.3.2/src/org/apache/cxf/ws/policy/PolicyEngineImpl.java:718]: Liberty change flag?

File: dev/com.ibm.ws.org.apache.cxf.cxf.rt.ws.policy.3.2/src/org/apache/cxf/ws/policy/PolicyEngineImpl.java [hunk 20]
```diff
@@ -694,12 +706,18 @@ public boolean supportsAlternative(Collection<? extends PolicyComponent> alterna
...
+                   LOG.fine("Not policy assertion, returning false");
```
Comment (jim-krueger) [inline dev/com.ibm.ws.org.apache.cxf.cxf.rt.ws.policy.3.2/src/org/apache/cxf/ws/policy/PolicyEngineImpl.java]: Liberty change flag?

File: dev/com.ibm.ws.org.apache.cxf.cxf.rt.ws.policy.3.2/src/org/apache/cxf/ws/policy/PolicyVerificationOutInterceptor.java [hunk 21]
```diff
@@ -0,0 +1,82 @@
...
+        } catch (PolicyException e) {
+            LOG.fine("An exception was thrown when verifying that the effective policy for "
+                     + "this request was satisfied.  However, this exception will not result in "
+                     + "a fault.  The exception raised is: "
+                     + e.toString());
+            return;
+        }
+        LOG.fine("Verified policies for outbound message.");
+    }
+}
```
Comment (jim-krueger) [inline dev/com.ibm.ws.org.apache.cxf.cxf.rt.ws.policy.3.2/src/org/apache/cxf/ws/policy/PolicyVerificationOutInterceptor.java:82]: What is the Liberty change in this class?

File: dev/com.ibm.ws.org.apache.cxf.cxf.rt.wsdl.3.2/src/org/apache/cxf/wsdl/interceptors/BareOutInterceptor.java [hunk 22]
```diff
@@ -0,0 +1,88 @@
...
+    public BareOutInterceptor() {
+        super(Phase.MARSHAL);
+    }
+
+    public void handleMessage(Message message) {
+        Exchange exchange = message.getExchange();
+        BindingOperationInfo operation = exchange.getBindingOperationInfo();
+
+        if (operation == null) {
+	    LOG.finest("BareOutInterceptor: Operation is NULL, returning..."); // Liberty Change start
```
Comment (jim-krueger) [inline dev/com.ibm.ws.org.apache.cxf.cxf.rt.wsdl.3.2/src/org/apache/cxf/wsdl/interceptors/BareOutInterceptor.java:49]: I don't see a Liberty Change end.

**Review by jim-krueger — CHANGES_REQUESTED** [review]
Some questions and I think some changes.

File: dev/com.ibm.ws.jaxws.ejb_fat/publish/servers/com.ibm.ws.jaxws.ejb.fat.ejbwsprovider/bootstrap.properties (hunk 5 above)
Comment (sawadood) [reply dev/com.ibm.ws.jaxws.ejb_fat/publish/servers/com.ibm.ws.jaxws.ejb.fat.ejbwsprovider/bootstrap.properties]: Good point, it's not necessary, I will remove this with another PR

File: dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/attachment/AttachmentUtil.java (hunk 6 above)
Comment (sawadood) [reply dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/attachment/AttachmentUtil.java]: I am not sure, but I will add a check for null to be safe.

File: dev/com.ibm.ws.jaxws.ejb_fat/publish/servers/com.ibm.ws.jaxws.ejb.fat.ejbwsprovider/bootstrap.properties (hunk 5 above)
Comment (sawadood) [reply dev/com.ibm.ws.jaxws.ejb_fat/publish/servers/com.ibm.ws.jaxws.ejb.fat.ejbwsprovider/bootstrap.properties]: Already removed this trace

File: dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/interceptor/OneWayProcessorInterceptor.java (hunk 8 above)
Comment (sawadood) [reply dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/interceptor/OneWayProcessorInterceptor.java:153]: No, this warning message exists in original CXF code.

File: dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/interceptor/MessageSenderInterceptor.java (hunk 7 above)
Comment (sawadood) [reply dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/interceptor/MessageSenderInterceptor.java]: I meant to add trace, but didn't find any good reason to add additional trace here, so will remove this LOG declaration.

File: dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/interceptor/StaxOutEndingInterceptor.java (hunk 9 above)
Comment (sawadood) [reply dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/interceptor/StaxOutEndingInterceptor.java]: Good catch, I will update to check for null here.

File: dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/staxutils/StaxUtils.java (hunk 11 above)
Comment (sawadood) [reply dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/staxutils/StaxUtils.java]: Will add Liberty Change flag here.

File: dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/staxutils/StaxUtils.java (hunk 13 above)
Comment (sawadood) [reply dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/staxutils/StaxUtils.java]: yes, will add Liberty Change flag here

File: dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/transport/ChainInitiationObserver.java (hunk 15 above)
Comment (sawadood) [reply dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/transport/ChainInitiationObserver.java]: yes, will add change flag

File: dev/com.ibm.ws.org.apache.cxf.cxf.rt.ws.policy.3.2/src/org/apache/cxf/ws/policy/PolicyEngineImpl.java (hunk 17 above)
Comment (sawadood) [reply dev/com.ibm.ws.org.apache.cxf.cxf.rt.ws.policy.3.2/src/org/apache/cxf/ws/policy/PolicyEngineImpl.java:691]: Not needed here as this is inside start/end change block; line 693 -> } // Liberty Change End

File: dev/com.ibm.ws.org.apache.cxf.cxf.rt.ws.policy.3.2/src/org/apache/cxf/ws/policy/PolicyEngineImpl.java (hunk 20 above)
Comment (sawadood) [reply dev/com.ibm.ws.org.apache.cxf.cxf.rt.ws.policy.3.2/src/org/apache/cxf/ws/policy/PolicyEngineImpl.java]: yes, will add change flag here

File: dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/workqueue/SynchronousExecutor.java (hunk 16 above)
Comment (sawadood) [reply dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/workqueue/SynchronousExecutor.java:47]: There is no liberty change here, was added by mistake.

File: dev/com.ibm.ws.org.apache.cxf.cxf.rt.ws.policy.3.2/src/org/apache/cxf/ws/policy/PolicyEngineImpl.java (hunk 18 above)
Comment (sawadood) [reply dev/com.ibm.ws.org.apache.cxf.cxf.rt.ws.policy.3.2/src/org/apache/cxf/ws/policy/PolicyEngineImpl.java:710]: Not needed here, it's inside Start/End block

File: dev/com.ibm.ws.org.apache.cxf.cxf.rt.ws.policy.3.2/src/org/apache/cxf/ws/policy/PolicyVerificationOutInterceptor.java (hunk 21 above)
Comment (sawadood) [reply dev/com.ibm.ws.org.apache.cxf.cxf.rt.ws.policy.3.2/src/org/apache/cxf/ws/policy/PolicyVerificationOutInterceptor.java:82]: Added liberty change flag to these 2 statements:

Comment (sawadood) [reply dev/com.ibm.ws.org.apache.cxf.cxf.rt.ws.policy.3.2/src/org/apache/cxf/ws/policy/PolicyVerificationOutInterceptor.java:82]: LOG.fine("AssertionInfoMap is null.");  // Liberty Change
LOG.fine("EffectivePolicy is null.");  // Liberty Change

File: dev/com.ibm.ws.org.apache.cxf.cxf.rt.wsdl.3.2/src/org/apache/cxf/wsdl/interceptors/BareOutInterceptor.java (hunk 22 above)
Comment (sawadood) [reply dev/com.ibm.ws.org.apache.cxf.cxf.rt.wsdl.3.2/src/org/apache/cxf/wsdl/interceptors/BareOutInterceptor.java:49]: missed end flag, will add it.

File: dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/interceptor/StaxOutInterceptor.java (hunk 10 above)
Comment (sawadood) [reply dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/interceptor/StaxOutInterceptor.java]: yes, will add null check to be safe

File: dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/staxutils/StaxUtils.java (hunk 14 above)
Comment (sawadood) [reply dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/staxutils/StaxUtils.java]: yes, will add Liberty change flag here

File: dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/staxutils/StaxUtils.java (hunk 12 above)
Comment (sawadood) [reply dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/staxutils/StaxUtils.java]: yes, will add Liberty change flag here

**Comment by sawadood** [comment]
#build - Hopefully last PB for this PR

**Review by jim-krueger — APPROVED** [review]

**Comment by sawadood** [comment]
#run-libby-bot

**Review by neuwerk — APPROVED** [review]
New changes look good.
